### ISSUE TITLE
Non standard external repositories break other rules -- cannot be overridden.

### ISSUE THREAD

**Issue by hsyed** [pubref/rules_kotlin#39]
Our workspace has most of the non dagger, non Kotlin transitive chain already defined. Calling the `Kotlin_repositories` macro breaks our build and adding omit params does not help. The external repositories defined by this library do not contain `//jar` packages.

**Comment by hsyed** [comment]
@pcj If you have some time could you look into this one, this one blocks us from using Kotlin. 

I've submitted a PR for a first pass on the Bazel IJ plugin.

**Comment by pcj** [comment]
Sorry for delay...  I'm confused though.  Which specific dependencies are we talking about that cannot be omitted by `kotlin_repositories`?  Each one appears to be guarded by an `omit_`.

**Comment by hsyed** [comment]
Slightly hazy on the details now. I think this repo does not follow standard bazel external repo jar targeting -- specifically the dependencies from here do not have a `//jar` target. So if I don't ommit my workspace cannot reach the `//jar`'s as the external name collides and this problem exists in the other direction as well.

**Comment by hsyed** [comment]
@pcj

here is an example: this is one of build files from this repo. It cannot pick up the guava jar from my workspace because it does not end with `//jar`.

```
java_library(
    name = "program",
    srcs = ["CommandLineProgram.java"],
    visibility = ["//visibility:public"],
    deps = ["@com_google_guava"], 
)
```

**Comment by hsyed** [comment]
@pcj so upon further investigation it was findbugs and error prone. All the other dependencies are named slightly differently from the maven naming scheme so they don't cause a problem. The pr fixes the problem for me and with it I don't need to use any omits anymore.

**Comment by pcj** [comment]
Closed by #42
